src/1_utils.js

```diff
@@ -61,6 +61,42 @@ utils.userPreferences = {
 	}
 };
 
+utils.generateDynamicBNCLink = function(branchKey, data) {
+	if (!branchKey && !data) {
+		return;
+	}
+	if (!data["data"] || typeof data["data"] !== "string") {
+		data = utils.cleanLinkData(data);
+	}
```

Comment:
You could just pass in the result of `utils.cleanLinkData(data)` instead of calling this function again.

Reply:
You should remove this conditional now since you are already passing in the result of the cleanLinkData function